src/utils/createErrorResponse.ts

```diff
@@ -1,13 +1,59 @@
-export function createErrorResponse(error: string | Error | unknown) {
-  let reason
+export interface Response {
+  v: '2'
+  notifications?: Notification[]
+  requestId: string
+  error?: ErrorData
+  products: {}
+}
+
+export interface Notification {
+  level: 'info' | 'warning' | 'error'
+  message: string
+}
+
+export interface ErrorData {
+  code?: 'Failed'
+  message: string
+}
+
+function errorToString(error: string | Error | unknown): string {
   try {
-    reason = typeof error === 'string' ? error : error instanceof Error ? error.message : String(error)
+    return typeof error === 'string' ? error : error instanceof Error ? error.message : String(error)
   } catch (e) {
-    reason = 'unknown'
+    return 'unknown'
+  }
+}
+
+function generateRandomString(length: number): string {
+  let result = ''
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length))
+  }
+  return result
+}
+
+function generateRequestUniqueId(): string {
```

Comment:
How do we plan to use this request id?

Reply:
This `requestId` has two purposes:
1- Separate the requests that reach and doesn't reach CoreAPI. With `cfi-` infix, a customer is able to tell that the response is generated by the CF Worker
2- Later on, if we can store logs, we can reach a specific request's logs by using `requestId` as an identifier.